ipaserver/install/ipa_migrate.py

```diff
@@ -31,8 +31,8 @@
 from ipaserver.install.ipa_migrate_constants import (
     DS_CONFIG, DB_OBJECTS, DS_INDEXES, BIND_DN, LOG_FILE_NAME,
     STRIP_OP_ATTRS, STRIP_ATTRS, STRIP_OC, PROD_ATTRS,
-    DNA_REGEN_VAL, DNA_REGEN_ATTRS, NIS_PLUGIN, IGNORE_ATTRS,
-    DB_EXCLUDE_TREES
+    DNA_REGEN_VAL, DNA_REGEN_ATTRS, IGNORE_ATTRS,
```

Comment:
The commit that removed NIS support was pushed to the master branch only, I believe you should keep NIS-related code in ipa-4-12 branch.
See https://github.com/freeipa/freeipa/pull/7336

Reply:
> The commit that removed NIS support was pushed to the master branch only, I believe you should keep NIS-related code in ipa-4-12 branch. See #7336

No problem, it's added back